### DISCUSSION TITLE
1.0.1

### DISCUSSION THREAD

**Discussion by benhillis** [microsoft/WSL#9189] [General]
* Fix a potential hang in wslservice.exe when launching a new session
* Make the `/tmp/.X11-unix` bind mount point read-only to prevent the socket from being deleted
* Don't fail to start if cgroup mounts fail [GH 8981]
* Add `E_FAIL` and `E_UNEXPECTED` to common error strings

<hr /><em>This discussion was created from the release <a href='https://github.com/microsoft/WSL/releases/tag/'>1.0.1</a>.</em>

**Comment by NotTheDr01ds** [comment]
> Make the `/tmp/.X11-unix` bind mount point read-only to prevent the socket from being deleted

Excellent!  I'm really hoping that will fix (most or all of) the remaining problems with it disappearing.  I know I changed mine on Artix (a non-Systemd distribution) a while back to re-mount as read-only with good success.

**Reply by cerebrate** [reply]
1.0.1 is a regression in this area for me, because the race condition is _still_ letting `/tmp` be mounted over the `/tmp/.X11-unix` bind mount, and since 1.0.1 masks `/lib/tmpfiles.d/x11.conf` rather than creating a backup symlink as 1.0 did, it also breaks the easy fix for the issue.

**Reply by NotTheDr01ds** [reply]
Well bummer.

**Reply by DemGiran** [reply]
Do you use Artix professionally
